packages/brick_graphql/lib/src/query_document_transformer.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:brick_graphql/graphql.dart';
+import 'package:brick_core/core.dart';
+import 'package:brick_graphql/src/runtime_graphql_definition.dart';
+import 'package:gql/ast.dart';
+
+class QueryDocumentTransformer<_Model extends GraphqlModel> {
+  final GraphQLAdapter adapter;
+
+  final List<GraphqlArgument> arguments;
+
+  /// Generates a document based on the [query]
```

Comment:
Might be worth adding where this definition of the document node came from
```suggestion
  /// Generates a document based on the [query], which is an ast representation of a Graphql Operation
  https://github.com/gql-dart/gql/blob/b761664f75834e4487a8c1d6f7bfd9a1c84adf5e/gql/lib/src/ast/ast.dart
```